gcp/api/server.py

```diff
@@ -518,17 +518,43 @@ def to_response(b):
     context.service_context.abort(grpc.StatusCode.INVALID_ARGUMENT,
                                   'Invalid query.')
 
+  # Asynchronously retrieve computed aliases and related ids here
+  # to prevent significant query time increase for packages with
+  # numerous vulnerabilities.
+  if include_details:
+    aliases = []
+    related = []
+    for bug in bugs:
+      aliases.append(osv.get_aliases_async(bug.id))
+      related.append(osv.get_related_async(bug.id))
+
+    for i, alias in enumerate(aliases):
+      alias_group = yield alias
+      if not alias_group:
+        continue
+      alias_ids = sorted(list(set(alias_group.bug_ids) - {bugs[i].id}))
+      bugs[i].aliases[:] = alias_ids
+      modified_time = bugs[i].modified.ToDatetime()
+      modified_time = max(alias_group.last_modified, modified_time)
+      bugs[i].modified.FromDatetime(modified_time)
+
+    for i, related_ids in enumerate(related):
+      related_bug_ids = yield related_ids
+      bugs[i].related[:] = sorted(
+          list(set(related_bug_ids + list(bugs[i].related))))
+
   if next_page_token:
     next_page_token = next_page_token.urlsafe()
     logging.warning('Page size limit hit, response size: %s', len(bugs))
 
   return bugs, next_page_token
 
 
-def bug_to_response(bug, include_details=True):
+def bug_to_response(bug, include_details=True, include_alias=False):
```

Comment:
Whenn would include_details != include_alias? Can we just consolidate these two into a single flag?

Reply:
The `include_alias` flag here is for use in `bug.to_vulnerability()` function. It only indicates whether we want to retrieve aliases within that function. If `include_alias` is false, aliases may be retrieved asynchronously through other methods.
For example, when we run `query_by_version/package` query, we want the `bug.to_vulnerability()` function to include the source but not the aliases. So we can retrieve the aliases for those vulnerabilities asynchronously afterward, making the query more efficient.(include_source=True, include_alias=False)
But when we use `GetVulnById()` function, we only get one vulnerability at a time. So we can just get everything we need from the `bug.to_vulnerability()` function. (include_source=True, include_alias=True)
I will add some comments here to make it clearer.